gloo/types.h

```diff
@@ -56,17 +56,27 @@ struct __attribute__((__aligned__(2))) float16 {
     return x == res.x;
   }
 #ifdef __CUDA_ARCH__
-  float16(half h) {
+  float16(__half h) {
```

Comment:
I think half and __half are interchangeable. cuda_fp16.h has the following code:

```
#ifndef CUDA_NO_HALF
typedef __half half;
typedef __half2 half2;
#endif /*CUDA_NO_HALF*/
```

Should we stick to one for consistency? I vote for half. :)